### ISSUE TITLE
Having only one CNAM for all orchestrator servers 

### ISSUE THREAD

**Issue by tapuhi** [openark/orchestrator#185]
Hi
Let's say we have 3 orchestrator servers, at any moment I have one active orchestrator app. I want to have only one management CNAM. In the amazon ELB I can only add basic http checks (also port checks and also a given html page can be checked)  .
Any other idea how to have only one CNAM for all orchestrator servers?
We can maybe use consule for that. 
BR

**Comment by shlomi-noach** [comment]
@tapuhi what you'd want to get is this: https://github.com/github/orchestrator/pull/183/commits/3f84a4dee3042a9f744dbdfb36289bd9de7e675b

which right now is a commit within an open PR, but I push that commit with a new branch and merge on `master`.

`/leader-check` is what your LB would use to test whether the node is active or not. 

if you can use the CNAME for the amazon ELB (is this possible?) I believe that solves your problem.

**Comment by tapuhi** [comment]
Thank you so much, 
Yes I can indeed use the CNAME for the amazon ELB.

**Comment by shlomi-noach** [comment]
don't close this yet, since `/leader-check` is not yet available to you.

**Comment by shlomi-noach** [comment]
Actually, you don't need `/leader-check` at this time. Any backend will do; just use the existing `/lb-check`, and your LB can do round robin or any other algorithm to pick the backend server.

**Comment by tapuhi** [comment]
Sorry I closed the issue. 
Anyway when will be the leader_check be available for us  ?

**Comment by shlomi-noach** [comment]
Please see my previous comment.

**Comment by tapuhi** [comment]
I am using authentication for orchestrator and /lb-check requires authentication . Is there anyway I can provide those in the URL ? 
Anyway I will need to know the leader for further connecting this elb to other systems I am integrating orchestrator with.

**Comment by shlomi-noach** [comment]
re: authentication: what kind of authentication? There are multiple options. If `Basic`, then no, there is no way to exclude the `/lb-check` path from authentication -- though you _can_ check `myusername:mypassword@myorchestrator:3000/api/lb-check`

re: leader - can you please elaborate what you need the leader identity for? You should be good to connect to any of the orchestrator hosts, equally.

**Comment by tapuhi** [comment]
Hi Shlomi,
   First of all I am happy to announce Orchestrator (you can add it to the users page :-) )  is fully in use for all mysql clusters in WIX. 
   I am building the monitors for orchestrator and want to test wether I have 2 leaders in case of split brain/network split and  for that I will need the leader test, I can use direct SQL queries but I truly prefer the api .  
  Would like to hear about other use cases of the leader check. 
    For all the rest you're right, I had a misunderstanding on how orchestrator leaders works (after reading an playing around I understood much better) . 

Thanks For the great product. 
Next step - Full integration with consul and ProxySQL

**Comment by shlomi-noach** [comment]
> First of all I am happy to announce Orchestrator (you can add it to the users page :-) ) is fully in use for all mysql clusters in WIX.

Thank you!

> for all mysql clusters in WIX.
I am building the monitors for orchestrator and want to test wether I have 2 leaders in case of split brain/network split and for that I will need the leader test, I can use direct SQL queries but I truly prefer the api .

Fair. I can deliver the leader check this week. You'll need to pull `master` though, I'm not going to create an official release this week.

> Next step - Full integration with consul and ProxySQL

Looking forward to hear about this work.

**Comment by shlomi-noach** [comment]
`/api/leader-check` is now available via merged https://github.com/github/orchestrator/pull/193

**Comment by tapuhi** [comment]
Thanks allot!
When are you going to GA this feature ?

**Comment by shlomi-noach** [comment]
@tapuhi unsure. There's a bunch of changes in `master` that need to be carefully announced.

**Comment by tapuhi** [comment]
Last question for this one, 
Is there a short tutorial how to build orchestrator from source ?
I tried the build.sh provided but not sure how to set GOPATH and GOROOT or any other env needed.

BTW, here is another reason why I need to know the elected node. I am using ELB upon nginx now (for the auth, which works good , thanks ) the problem is that each physical server I land on have a different cookie so I need basically to authenticate on each click , I am trying to figure out if there is a nginx solution for that.

**Comment by shlomi-noach** [comment]
> Is there a short tutorial how to build orchestrator from source ?

https://github.com/github/orchestrator/blob/master/docs/developers.md

but also try `./script/build` which runs an isolated build; that's the one used by CI. That would be easiest!

**Comment by shlomi-noach** [comment]
> BTW, here is another reason why I need to know the elected node. I am using ELB upon nginx now (for the auth, which works good , thanks ) the problem is that each physical server I land on have a different cookie so I need basically to authenticate on each click , I am trying to figure out if there is a nginx solution for that.

you mean your ELB does not support stickey sessions?

**Comment by tapuhi** [comment]
I did in a mistake in the amazon ELB config. I created a TCP elb, which does not support stickey, instead of HTTP elb. Now it works and Authentication works just fine now.

THanks

**Comment by tapuhi** [comment]
Hi All, 
   I've opened a case in consul google group and stackoverflow:

https://stackoverflow.com/questions/45021012/consul-template-with-large-number-of-keys-in-kv
https://groups.google.com/forum/#!topic/consul-tool/XYsUsOBASsU

The template proposed there is a first draft, which already works end to end (orchestrator <-> proxysql).

Just to clarify, this same template was already in use in a test env with few clusters so I did not encountered this issue there. 

Would appreciate if any of you guys can have a look on this.

BR
